packages/core/src/messageformat.ts

```diff
@@ -61,12 +70,18 @@ export interface MessageFormatOptions<
    */
   customFormatters?: {
     [key: string]:
-      | CustomFormatter
+      | DefaultCustomFormatter
       | {
-          formatter: CustomFormatter;
+          formatter: DefaultCustomFormatter;
           arg?: 'string' | 'raw' | 'options';
           id?: string;
           module?: string;
+        }
+      | {
+          formatter: LocaleModuleCustomFormatter;
+          arg?: 'string' | 'raw' | 'options';
+          id?: string;
+          module?: (locale: string) => string;
```

Comment:
some of these keys could be deduped, but they were small enough i just dupe'd em (e.g. `arg`, `id`)